packages/medusa/src/services/fulfillment.ts

```diff
@@ -83,15 +93,19 @@ class FulfillmentService extends BaseService {
 
   /**
    * Retrieves the order line items, given an array of items.
-   * @param {Order} order - the order to get line items from
-   * @param {{ item_id: string, quantity: number }} items - the items to get
-   * @param {function} transformer - a function to apply to each of the items
+   * @param order - the order to get line items from
+   * @param items - the items to get
+   * @param transformer - a function to apply to each of the items
    *    retrieved from the order, should return a line item. If the transformer
    *    returns an undefined value the line item will be filtered from the
    *    returned array.
-   * @return {Promise<Array<LineItem>>} the line items generated by the transformer.
+   * @return the line items generated by the transformer.
    */
-  async getFulfillmentItems_(order, items, transformer) {
+  async getFulfillmentItems_<T>(
```

Comment:
question: Why is that method template?

Reply:
due to the return type being dependent on whatever `transform` function is passed to the method, however I have removed the transform method parameter and just referenced the method directly.